src/mapservutil.c

```diff
@@ -1993,6 +1988,23 @@ int msCGIDispatchQueryRequest(mapservObj *mapserv) {
     /* finally execute the query */
     if (msExecuteQuery(mapserv->map) != MS_SUCCESS)
       return MS_FAILURE;
+
+    /* catch empty result set when web->empty is set (#6907) */
+    if (mapserv->map->web.empty != NULL ||
+        CPLGetConfigOption("MS_EMPTY_URL", NULL) != NULL) {
+      int n = 0;
+      for (int i = 0; i < mapserv->map->numlayers; i++) { // count results
+        if (mapserv->map->layers[i]->resultcache) {
+          n += mapserv->map->layers[i]->resultcache->numresults;
+        }
+      }
+      if (n == 0) {
+        /* note: this error message will not be displayed */
```

Comment:
So this error will only appear in the logs, and only in the case when the WEB EMPTY pararameter has been set?
Sounds good - just wanted to confirm the logic, and this means the issue #5706 can be avoided.

Reply:
If it's logged then it will only be when WEB EMPTY or the config MS_EMPTY_URL values are set.